### ISSUE TITLE
Params naming

### ISSUE THREAD

**Issue by p3jawors** [project-rig/nengo_spinnaker#108]
Ran into a problem with the most recent version of Nengo about paramter naming, it was resolved by using the following code in `config.py` (adding `name=`):

``` python

def add_spinnaker_params(config):
    """Add SpiNNaker specific parameters to a configuration object."""
    # Add simulator parameters
    config.configures(Simulator)

    config[Simulator].set_param("placer", CallableParameter(name="placer", default=par.place))
    config[Simulator].set_param("placer_kwargs", DictParam(name="placer_kwargs", default={}))

    config[Simulator].set_param("allocater",
                                CallableParameter(name="allocater", default=par.allocate))
    config[Simulator].set_param("allocater_kwargs",
                                DictParam(name="allocater_kwargs", default={}))

    config[Simulator].set_param("router", CallableParameter(name="router", default=par.route))
    config[Simulator].set_param("router_kwargs", DictParam(name="router_kwargs", default={}))

    config[Simulator].set_param("node_io", Parameter(name="node_io", default=Ethernet))
    config[Simulator].set_param("node_io_kwargs", DictParam(name="node_io_kwargs", default={}))

    # Add function_of_time parameters to Nodes
    config[nengo.Node].set_param("function_of_time", BoolParam(name="function_of_time", default=False))
    config[nengo.Node].set_param("function_of_time_period",
                                 NumberParam(name="function_of_time_kwargs", default=None, optional=True))

    # Add profiling parameters to Ensembles
    config[nengo.Ensemble].set_param("profile", BoolParam(name="profile", default=False))
    config[nengo.Ensemble].set_param("profile_num_samples",
                                     NumberParam(name="profile_num_samples", default=None, optional=True))
```

**Comment by tcstewar** [comment]
Some quick context for this:

We made a change in how nengo parameters are defined, and we completely screwed up and didn't notice that it's horribly non-backwards-compatible.  It is a small change (just adding the name parameter to the Param constructor), but making it means that it's awkward to write code that works for both the current master and previous versions of nengo.  Which is a problem.

I worked around this in nengo_gui with this approach:

https://github.com/nengo/nengo_gui/commit/811a00cc64fc0593b8a3ad478a4211e68a008fcc

Basically it tries to make the object with the new syntax, and then falls back onto the old syntax if that fails.  I think it'd make sense to do something similar here.  I could take a stab at doing that, if that'd be useful.
